cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -271,17 +271,16 @@ Result<S3Options> S3Options::FromUri(const Uri& uri, std::string* out_path) {
     options.ConfigureDefaultCredentials();
   }
 
-  auto it = options_map.find("region");
-  if (it != options_map.end()) {
-    options.region = it->second;
-  }
-  it = options_map.find("scheme");
-  if (it != options_map.end()) {
-    options.scheme = it->second;
-  }
-  it = options_map.find("endpoint_override");
-  if (it != options_map.end()) {
-    options.endpoint_override = it->second;
+  for (const auto& kv : options_map) {
```

Comment:
Is it guaranteed that all of these query param values can only have safe characters and don't need to be URI decoded? (Related: I don't see docs for all of these so I don't know how to use them or what valid values for them are.)

Reply:
They are automatically decoded already by the `Uri` class.